hydra-plutus/src/Hydra/Contract/Util.hs

```diff
@@ -47,3 +48,15 @@ mustNotMintOrBurn TxInfo{txInfoMint} =
     isZero txInfoMint
 {-# INLINEABLE mustNotMintOrBurn #-}
 
+
+
+infix 4 ===!
+
+-- | Checks for exact exuality between two serialized values
+-- Equality on value is very memory intensive as it's defined on associative
+-- lists and Map equality is implemented. Instead we can be more strict and
+-- require EXACTLY the same value and compare using the serialised bytes.
+(===!) :: Value -> Value -> Bool
+(===!) val val' =
+  serialiseData (toBuiltinData val) == serialiseData (toBuiltinData val')
```

Comment:
Have you also tried comparing the hashes or doing an explicit loop through the value elements?

Asked differently, do we know if this is the cheapest way to do it?

Reply:
I haven't tried it but will do and compare the results.

Reply:
I added the logbook entry [here](https://github.com/input-output-hk/hydra/wiki/Logbook-2023-H1#sb-on-alternative-implementation-for-scrict-equality-assocmap-check) and it shows that serializing data is our best bet. Perhaps I should try alternative implementation where I don't rely on plutusTx functions like `flattenValue` and `valueOf` but I didn't see anything that would make them slower then they need be.